src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/ToolRestoreUtility.cs

```diff
@@ -64,7 +64,7 @@ public static PackageSpec GetSpec(string projectFilePath, string id, VersionRang
                             ProjectReferences = { }
                         }
                     },
-                    ProjectWideWarningProperties = projectWideWarningProperties
+                    ProjectWideWarningProperties = projectWideWarningProperties ?? new WarningProperties()
```

Comment:
I'd imagine this is inconsistent with how tools are handled on the commandline. 

On the commandline they are not even read. So maybe you need to address that.

Reply:
In commandline, the restore metadata's [default value](https://github.com/NuGet/NuGet.Client/blob/e35b70a6ba06d3b7249edee96650a4ed82192070/src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreMetadata.cs#L113) is used, which is the same outcome.

Reply:
Cool. Makes sense. 

Since it's either null or empty, it probably wasn't an issue before because it was never written down.